tests/test_distribuited/test_nats.py

```diff
@@ -0,0 +1,109 @@
+import asyncio
+import socketserver
+import concurrent.futures
+import subprocess
+
+from dffml.util.asynctestcase import AsyncTestCase
+
+
+from dffml.distribuited.orchestrator import NatsWorkerNode, NatsWorkerNodeConfig
+from dffml import DataFlow
+from dffml_feature_git.feature.operations import (
+    check_if_valid_git_repository_URL,
+)
+from dffml.operation.output import GetSingle
+from dffml.operation.db import (
+    DatabaseQueryConfig,
+    db_query_create_table,
+)
+
+import os
+import tempfile
+import asyncio
+
+from dffml.distribuited.orchestrator import (
+    NatsOrchestratorNode,
+    NatsOrchestratorNodeConfig,
+    NatsWorkerNode,
+    NatsWorkerNodeConfig,
+)
+from dffml import DataFlow
+from dffml_feature_git.feature.operations import (
+    check_if_valid_git_repository_URL,
+)
+from dffml.operation.output import GetSingle
+from dffml.db.sqlite import SqliteDatabase, SqliteDatabaseConfig
+from dffml.operation.db import (
+    DatabaseQueryConfig,
+    db_query_create_table,
+)
+
+
+class TestNatsOrchestrator(AsyncTestCase):
+    async def setUp(self):
+        with socketserver.TCPServer(("localhost", 0), None) as s:
+            free_port = s.server_address[1]
+
+        self.server_addr = f"0.0.0.0:{free_port}"
+
+        self.nats_proc = subprocess.Popen(
+            ["nats-server", "-p", str(free_port),],
```

Comment:
Let's make sure we have NATs choose the port

https://github.com/nats-io/nats-server/issues/1097#issuecomment-637232781

```suggestion
            ["nats-server", "-p", "-1"],
```